src/MongoDB/Manager.c

```diff
@@ -669,11 +670,46 @@ static PHP_METHOD(Manager, startSession)
 		return;
 	}
 
-	if (options && php_array_exists(options, "causalConsistency")) {
+	if (options && php_array_existsc(options, "causalConsistency")) {
 		cs_opts = mongoc_session_opts_new();
 		mongoc_session_opts_set_causal_consistency(cs_opts, php_array_fetchc_bool(options, "causalConsistency"));
 	}
 
+	if (options && php_array_existsc(options, "defaultTransactionOptions")) {
+		zval* txn_options = php_array_fetch_array(options, "defaultTransactionOptions");
+		mongoc_transaction_opt_t* txn_opts = NULL;
+
+		/* Thrown exception and return if the defaultTransactionOptions is not an array */
+		if (!txn_options) {
+			phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected \"defaultTransactionOptions\" option to be an array");
```

Comment:
For consistency with other unexpected type exceptions, can we change `php_array_fetch_array` above to `php_array_fetchc` and then use the following here?

```c
if (Z_TYPE_P(txn_options) != IS_ARRAY) {
	phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected \"defaultTransactionOptions\" option to be array, %s given", PHONGO_ZVAL_CLASS_OR_TYPE_NAME_P(txn_options));
}
```

Reply:
OK